Source/Core/Core/HW/Memmap.cpp

```diff
@@ -154,15 +174,10 @@ struct LogicalMemoryView
 //
 // Dolphin doesn't emulate the difference between cached and uncached access.
 //
-// TODO: The actual size of RAM is REALRAM_SIZE (24MB); the other 8MB shouldn't
+// TODO: The actual size of RAM is g_ram_size_real (24MB); the other 8MB shouldn't
 // be backed by actual memory.
 // TODO: Do we want to handle the mirrors of the GC RAM?
-static PhysicalMemoryRegion physical_regions[] = {
-    {&m_pRAM, 0x00000000, RAM_SIZE, PhysicalMemoryRegion::ALWAYS},
-    {&m_pL1Cache, 0xE0000000, L1_CACHE_SIZE, PhysicalMemoryRegion::ALWAYS},
-    {&m_pFakeVMEM, 0x7E000000, FAKEVMEM_SIZE, PhysicalMemoryRegion::FAKE_VMEM},
-    {&m_pEXRAM, 0x10000000, EXRAM_SIZE, PhysicalMemoryRegion::WII_ONLY},
-};
+static PhysicalMemoryRegion physical_regions[4];
```

Comment:
```suggestion
static std::array<PhysicalMemoryRegion, 4> physical_regions;
```

(`<array>` may need to be included)

Reply:
The original implementation did not use std::array.  What benefits does this bring to the table?  Is this just a general improvement that we ought to do while editing Memmap.cpp?

Reply:
If you're already changing the line, might as well make it better if it's trivial to do so, y'know? Or do you want a general list of reasons why std::array is better than a C-style array?

Reply:
Moreso the latter.  I'm still a learning a lot about C++, and have never used std::array before.

Reply:
- [std::array](https://en.cppreference.com/w/cpp/container/array) cannot have implicit array to pointer decay, It needs to be explicitly done by using it's `.data()` member function where necessary.

- `std::array` has a `size()` member function, so you can query the size of an array without needing to use `sizeof(array[0]) / sizeof(array)` nonsense.

- It's a wrapper around a C array. This means that implementations can optionally provide bounds checking in debug builds (controllable via a define) to catch out of bounds accesses that may occur across refactoring (currently all three main implementations—MSVC, libstdc++, and libc++—provide this).

TL;DR it's a more strongly-typed and type-safe version of a built-in array.